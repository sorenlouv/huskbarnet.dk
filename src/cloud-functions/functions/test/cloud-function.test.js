import jest from 'jest';
import res from './response.json';
import {
  formatReminders,
  getDates,
  getReminderMessages
} from '../../../cloud-functions/functions/getRemindersHelpers';

describe('formatReminders', () => {
  it('should format reminders correctly', () => {
    expect(formatReminders(res)).toMatchSnapshot();
  });
});

describe('getDates', () => {
  it('should return 7 dates ', () => {
    expect(getDates('2015-04-01')).toEqual([
      '2015-01-01',
      '2014-11-01',
      '2014-04-01',
      '2014-01-01',
      '2011-04-01',
      '2010-04-01',
      '2003-04-01'
    ]);
  });
});

describe('getReminderMessages', () => {
  it('It should format messages correctly', () => {
    const adminMock = {};
    adminMock.database = () => adminMock;
    adminMock.ref = () => adminMock;
    adminMock.orderByChild = () => adminMock;
    adminMock.equalTo = () => adminMock;
    adminMock.once = () => adminMock;
    adminMock.then = cb => cb({ val: () => res });

    return expect(
      getReminderMessages('2017-12-22', adminMock)
    ).resolves.toEqual();
  });
});
