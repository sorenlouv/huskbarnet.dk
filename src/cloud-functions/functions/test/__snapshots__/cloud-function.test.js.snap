// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatReminders should format reminders correctly 1`] = `
Array [
  Object {
    "dateOfBirth": "2017-12-22",
    "email": "myemail+xtra@gmail.com",
    "emails": Object {
      "-LA9um4pV5zg_Ac9cYgc": "myemail+xtra@gmail.com",
      "auth": "myemail@gmail.com",
    },
    "gender": "male",
    "name": "Lars",
  },
  Object {
    "dateOfBirth": "2017-12-22",
    "email": "myemail@gmail.com",
    "emails": Object {
      "-LA9um4pV5zg_Ac9cYgc": "myemail+xtra@gmail.com",
      "auth": "myemail@gmail.com",
    },
    "gender": "male",
    "name": "Lars",
  },
  Object {
    "dateOfBirth": "2016-01-25",
    "email": "myemail+xtra@gmail.com",
    "emails": Object {
      "-LA9um4pV5zg_Ac9cYgc": "myemail+xtra@gmail.com",
      "auth": "myemail@gmail.com",
    },
    "gender": "male",
    "name": "Søren",
  },
  Object {
    "dateOfBirth": "2016-01-25",
    "email": "myemail@gmail.com",
    "emails": Object {
      "-LA9um4pV5zg_Ac9cYgc": "myemail+xtra@gmail.com",
      "auth": "myemail@gmail.com",
    },
    "gender": "male",
    "name": "Søren",
  },
  Object {
    "dateOfBirth": "2016-08-15",
    "email": "myemail+xtra@gmail.com",
    "emails": Object {
      "-LA9um4pV5zg_Ac9cYgc": "myemail+xtra@gmail.com",
      "auth": "myemail@gmail.com",
    },
    "gender": "female",
    "name": "Sille",
  },
  Object {
    "dateOfBirth": "2016-08-15",
    "email": "myemail@gmail.com",
    "emails": Object {
      "-LA9um4pV5zg_Ac9cYgc": "myemail+xtra@gmail.com",
      "auth": "myemail@gmail.com",
    },
    "gender": "female",
    "name": "Sille",
  },
]
`;
